compiler/include/files.h

```diff
@@ -53,7 +53,8 @@ const char* objectFileForCFile(const char* cfile);
 
 const char* genIntermediateFilename(const char* filename);
 
-void openCFile(fileinfo* fi, const char* name, const char* ext = NULL);
+void openCFile(fileinfo* fi, const char* name, const char* ext = NULL,
```

Comment:
I don't think you should modify openCFile in this way. Instead, can you either:
 * call openfile and closefile directly, or
 * add something like openOutputFile / openOtherFile that calls openfile/closefile ?

Also I just noticed that closeCFile does not check the return code from fclose. Could you modify it to call closefile instead of fclose, while you're here?

Reply:
Sorry, should have made it more clear: I was viewing this change as temporary - I'll make my own version for when I add the library directory and then all the generated library files will use that version instead.

Reply:
But I'm happy to make it now and adjust that new function when I add the library directory functionality